### PR TITLE
Support fixed features in Service API

### PR DESCRIPTION
**Pull request by AngelFP**
Addresses https://github.com/facebook/Ax/issues/746 (also in the wishlist https://github.com/facebook/Ax/issues/566).

As the title implies, this PR adds the possibility of specifying some `ObservationFeatures` as `fixed_features` in `AxClient.get_next_trial` and `AxClient.get_next_trials`. From my understanding, this is currently only possible with the developer API.

This is an option we would like to expose in [optimas](https://github.com/optimas-org/optimas), where we use the Service API for most of the BO generators.

### CODE REVIEW

**Comment by AngelFP** [comment]
Hi all,

has there been any update on this? We're currently having to use a [custom subclass](https://github.com/optimas-org/optimas/blob/706f80af8fecc71455d15531e7ebdc58afa78adb/optimas/generators/ax/service/custom_ax.py) of `AxClient` in order to have this feature, so it would be great if it could be merged :D (it would also address https://github.com/facebook/Ax/issues/1951).

Feel free to let me know if there's anything I should change in how I implemented the feature.

**Comment by sgbaird** [comment]
Seconded on @AngelFP's comment! I've been itching to implement a fixed feature option within https://honegumi.readthedocs.io/ to better serve the researchers in my circle (coming back to this because I just wrapped up a long conversation about it with a colleague), but it's going to be much more straightforward to do so with this PR merged.

@Cesar-Cardoso and co., does the Facebook Internal - Linter check need to be fixed? Non- meta employees can't [view details on this](https://github.com/facebook/Ax/pull/2015/checks?check_run_id=19231251410).

File: ax/service/ax_client.py [hunk 1]
```diff
@@ -1765,6 +1787,8 @@ def _gen_new_generator_run(self, n: int = 1) -> GeneratorRun:
                 parameters={}, trial_index=self._get_last_completed_trial_index()
             )
         )
+        if fixed_features:
+            fixed_feats.update_features(fixed_features)
```
Comment (sgbaird) [inline ax/service/ax_client.py:1791]: @Cesar-Cardoso This is really the only thing that's changing. The rest is plumbing to be able to pass `fixed_features` down to `_gen_new_generator_run(...)`.

**Comment by Cesar-Cardoso** [comment]
Hi folks! Apologies for the slow response from my part. Support for fixed features is being added now in https://github.com/facebook/Ax/pull/2372. Only difference being the use of `FixedFeatures` instead of `ObservationFeatures`, as `ObservationFeatures` isn't meant to be used as input for the API. We also had some internal discussion about using `TParameterization` directly instead of `FixedFeatures`, but decided against it.

**Comment by AngelFP** [comment]
Hi @Cesar-Cardoso , great to see this done! Thanks for looking into it.

**Comment by sgbaird** [comment]
I'll try to put a MWE up soon, but a starting point might be https://github.com/facebook/Ax/blob/18d08424c4a11ca8ed9a50b760c87826e886be22/ax/service/tests/test_ax_client.py#L2838-L2863

**Pull request closed without merging**